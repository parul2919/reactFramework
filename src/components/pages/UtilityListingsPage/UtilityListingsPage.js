import React from "react";
import UtilityListings from "../../template/UtilityListings/UtilityListings";

const wrapperStyles = {
  maxWidth: "100%",
  margin: "auto",
};

const UtilityListingsPage = () => (
  <div style={wrapperStyles}>
    <UtilityListings />
  </div>
);

export default UtilityListingsPage;
