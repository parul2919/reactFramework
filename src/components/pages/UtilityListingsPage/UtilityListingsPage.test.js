import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UtilityListingsPage from "./UtilityListingsPage";

describe("<Component /> spec", () => {
  it("renders the component", () => {
    const view = render(
      <Router>
        <UtilityListingsPage />
      </Router>
    );
    expect(view).toMatchSnapshot();
  });
});
