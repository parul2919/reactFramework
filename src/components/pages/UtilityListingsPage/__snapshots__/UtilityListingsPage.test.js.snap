// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="max-width: 100%; margin: auto;"
      >
        <div
          class="sc-gKXOVf kysQOZ dummy-template"
        >
          <header
            class="header sc-eCYdqJ lkPLVQ"
          >
            <nav
              class="navbar  sc-hKMtZM jzgEMa"
            >
              <a
                class="navbar-brand "
                href="/"
              >
                <img
                  alt="logo-img"
                  src="test-file-stub"
                  width="150"
                />
              </a>
              <div
                class="sc-gsnTZi gXHhKa collapse navbar-collapse menu-items"
              >
                <ul
                  class="navbar-nav ml-auto main-menu "
                >
                  <li
                    class="active dropdown-hide"
                  >
                    <a
                      class="nav-link"
                      href="/"
                      target="_self"
                    >
                      Tech Stacks
                    </a>
                  </li>
                  <li
                    class="inactive dropdown-hide"
                  >
                    <a
                      class="nav-link"
                      href="/component-library"
                      target="_self"
                    >
                      Component Library
                    </a>
                  </li>
                  <li
                    class="inactive dropdown-hide"
                  >
                    <a
                      class="nav-link"
                      href="/project-structure"
                      target="_self"
                    >
                      Project Structure
                    </a>
                  </li>
                  <li
                    class="inactive dropdown-hide"
                  >
                    <a
                      class="nav-link"
                      href="/utilities"
                      target="_self"
                    >
                      Utilities
                    </a>
                  </li>
                  <li
                    class="inactive dropdown-hide"
                  >
                    <a
                      class="nav-link"
                      href="/testNav"
                      target="_self"
                    >
                      TestNav
                    </a>
                    <div
                      class="undefined collapse navbar-collapse sub-menu"
                    >
                      <ul
                        class="navbar-nav ml-auto sub-menu-1 "
                      >
                        <li
                          class="inactive dropdown-hide"
                        >
                          <a
                            class="nav-link"
                            href="/form"
                            target="_self"
                          >
                            FormEx
                          </a>
                        </li>
                        <li
                          class="inactive dropdown-hide"
                        >
                          <a
                            class="nav-link"
                            href="/2"
                            target="_self"
                          >
                            Subnav2
                          </a>
                          <div
                            class="undefined collapse navbar-collapse sub-menu"
                          >
                            <ul
                              class="navbar-nav ml-auto sub-menu-2 "
                            >
                              <li
                                class="inactive dropdown-hide"
                              >
                                <a
                                  class="nav-link"
                                  href="/3"
                                  target="_self"
                                >
                                  NestedNav
                                </a>
                              </li>
                            </ul>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </li>
                </ul>
              </div>
            </nav>
          </header>
          <section
            class="techStack sc-jSMfEi fCjbVs wrapper-std"
          >
            <div
              class="techStack-list"
            >
              <h2>
                All the Utilities listing can be found here inside Common folder ie, src :: common
              </h2>
              <h3>
                1. JS Utilieis
              </h3>
              <ul>
                <li>
                  <b>
                    convertCashing
                  </b>
                   - This function allows to chnage casing of given "string without spaces" to the desired one
                </li>
                <li>
                  <b>
                    dataFetcher
                  </b>
                   - Allows to make api calls using browser fetch api
                </li>
                <li>
                  <b>
                    debounceFunc
                  </b>
                   - This function allows to limit or hinder the execution of a function based on a given condition.
                </li>
                <li>
                  <b>
                    ellipsis
                  </b>
                   - This function allows to tructate string with defined character limit and adds ellipsis at end
                </li>
                <li>
                  <b>
                    fileUpload
                  </b>
                   - Implements file upload by sending formData using browser fetch api
                </li>
                <li>
                  <b>
                    ga-init
                  </b>
                  - initializer of ga4 intance
                </li>
                <li>
                  <b>
                    generateRandom
                  </b>
                  - Allows to geterate random numbers based on defined upper & lower limit
                </li>
                <li>
                  <b>
                    getNameInitial
                  </b>
                  - This function allows to get name initials
                </li>
                <li>
                  <b>
                    hashing
                  </b>
                   - Allows to get hashed value of giving string eg, email
                </li>
                <li>
                  <b>
                    lcStorageUtil
                  </b>
                  - Allows to do local storage operations
                </li>
                <li>
                  <b>
                    removeArrDuplicate
                  </b>
                  - Allows to remove all the duplicate values from an array
                </li>
                <li>
                  <b>
                    repeatTimes
                  </b>
                  - Allows to do required operation x times
                </li>
                <li>
                  <b>
                    slugify
                  </b>
                  - Allows to slugify give string with hyphenated input
                </li>
                <li>
                  <b>
                    sorting
                  </b>
                  - This function allows array sorting
                </li>
                <li>
                  <b>
                    validations
                  </b>
                   - This function is having validation rules for form input
                </li>
                <li>
                  <b>
                    withStyle
                  </b>
                   - HOC : The component send styles in form of an object
                </li>
              </ul>
              <h3>
                2. Style Themes
              </h3>
              <ul>
                <li>
                  <b>
                    Theme
                  </b>
                   - Theme variables export can be find here like Font, color, media etc
                </li>
                <li>
                  <b>
                    fonts
                  </b>
                   - Defines font configuration variables
                   
                </li>
                <li>
                  <b>
                    colors
                  </b>
                   - Defines theme color configuration variables
                   
                </li>
                <li>
                  <b>
                    gridConfig
                  </b>
                   - Defines gridConfig configuration variables such as offset, gutterspaces etc
                </li>
                <li>
                  <b>
                    dimensions
                  </b>
                   - Defines mediaqueris configuration variables
                   
                </li>
                <li>
                  <b>
                    checkbox
                  </b>
                   - Enables custom style checkbox implementation
                   
                </li>
                <li>
                  <b>
                    buttons
                  </b>
                   - Predefined reusable button classes such as button--primary, button--disable etc
                </li>
                <li>
                  <b>
                    confirmPopOverStyle
                  </b>
                   - Specifies predefined common popover stylings
                </li>
              </ul>
              <h3>
                3. Style Css
              </h3>
              <ul>
                <li>
                  <b>
                    index.css
                  </b>
                   - Aggregator file of defined css files
                </li>
                <li>
                  <b>
                    normaliza.css
                  </b>
                   - Style file to reset default browser properties of HTML tags
                </li>
                <li>
                  <b>
                    utilClasses
                  </b>
                   - Style file contains common handy classes to put styling based on bootsrap class definitions
                </li>
                <li>
                  <b>
                    fonts.css
                  </b>
                  imports font family required for project other than browser fonts
                </li>
              </ul>
            </div>
          </section>
          <section
            class="footer sc-bczRLJ eRSPWO"
          >
            <nav
              style="max-width: 1440px; margin: auto;"
            >
              <a
                class="footer-logo"
                href="/"
              >
                <img
                  alt="logo-img"
                  src="test-file-stub"
                  width="150"
                />
              </a>
            </nav>
            <div
              class="disclaimer-text text-center"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="max-width: 100%; margin: auto;"
    >
      <div
        class="sc-gKXOVf kysQOZ dummy-template"
      >
        <header
          class="header sc-eCYdqJ lkPLVQ"
        >
          <nav
            class="navbar  sc-hKMtZM jzgEMa"
          >
            <a
              class="navbar-brand "
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
            <div
              class="sc-gsnTZi gXHhKa collapse navbar-collapse menu-items"
            >
              <ul
                class="navbar-nav ml-auto main-menu "
              >
                <li
                  class="active dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/"
                    target="_self"
                  >
                    Tech Stacks
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/component-library"
                    target="_self"
                  >
                    Component Library
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/project-structure"
                    target="_self"
                  >
                    Project Structure
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/utilities"
                    target="_self"
                  >
                    Utilities
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/testNav"
                    target="_self"
                  >
                    TestNav
                  </a>
                  <div
                    class="undefined collapse navbar-collapse sub-menu"
                  >
                    <ul
                      class="navbar-nav ml-auto sub-menu-1 "
                    >
                      <li
                        class="inactive dropdown-hide"
                      >
                        <a
                          class="nav-link"
                          href="/form"
                          target="_self"
                        >
                          FormEx
                        </a>
                      </li>
                      <li
                        class="inactive dropdown-hide"
                      >
                        <a
                          class="nav-link"
                          href="/2"
                          target="_self"
                        >
                          Subnav2
                        </a>
                        <div
                          class="undefined collapse navbar-collapse sub-menu"
                        >
                          <ul
                            class="navbar-nav ml-auto sub-menu-2 "
                          >
                            <li
                              class="inactive dropdown-hide"
                            >
                              <a
                                class="nav-link"
                                href="/3"
                                target="_self"
                              >
                                NestedNav
                              </a>
                            </li>
                          </ul>
                        </div>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
            </div>
          </nav>
        </header>
        <section
          class="techStack sc-jSMfEi fCjbVs wrapper-std"
        >
          <div
            class="techStack-list"
          >
            <h2>
              All the Utilities listing can be found here inside Common folder ie, src :: common
            </h2>
            <h3>
              1. JS Utilieis
            </h3>
            <ul>
              <li>
                <b>
                  convertCashing
                </b>
                 - This function allows to chnage casing of given "string without spaces" to the desired one
              </li>
              <li>
                <b>
                  dataFetcher
                </b>
                 - Allows to make api calls using browser fetch api
              </li>
              <li>
                <b>
                  debounceFunc
                </b>
                 - This function allows to limit or hinder the execution of a function based on a given condition.
              </li>
              <li>
                <b>
                  ellipsis
                </b>
                 - This function allows to tructate string with defined character limit and adds ellipsis at end
              </li>
              <li>
                <b>
                  fileUpload
                </b>
                 - Implements file upload by sending formData using browser fetch api
              </li>
              <li>
                <b>
                  ga-init
                </b>
                - initializer of ga4 intance
              </li>
              <li>
                <b>
                  generateRandom
                </b>
                - Allows to geterate random numbers based on defined upper & lower limit
              </li>
              <li>
                <b>
                  getNameInitial
                </b>
                - This function allows to get name initials
              </li>
              <li>
                <b>
                  hashing
                </b>
                 - Allows to get hashed value of giving string eg, email
              </li>
              <li>
                <b>
                  lcStorageUtil
                </b>
                - Allows to do local storage operations
              </li>
              <li>
                <b>
                  removeArrDuplicate
                </b>
                - Allows to remove all the duplicate values from an array
              </li>
              <li>
                <b>
                  repeatTimes
                </b>
                - Allows to do required operation x times
              </li>
              <li>
                <b>
                  slugify
                </b>
                - Allows to slugify give string with hyphenated input
              </li>
              <li>
                <b>
                  sorting
                </b>
                - This function allows array sorting
              </li>
              <li>
                <b>
                  validations
                </b>
                 - This function is having validation rules for form input
              </li>
              <li>
                <b>
                  withStyle
                </b>
                 - HOC : The component send styles in form of an object
              </li>
            </ul>
            <h3>
              2. Style Themes
            </h3>
            <ul>
              <li>
                <b>
                  Theme
                </b>
                 - Theme variables export can be find here like Font, color, media etc
              </li>
              <li>
                <b>
                  fonts
                </b>
                 - Defines font configuration variables
                 
              </li>
              <li>
                <b>
                  colors
                </b>
                 - Defines theme color configuration variables
                 
              </li>
              <li>
                <b>
                  gridConfig
                </b>
                 - Defines gridConfig configuration variables such as offset, gutterspaces etc
              </li>
              <li>
                <b>
                  dimensions
                </b>
                 - Defines mediaqueris configuration variables
                 
              </li>
              <li>
                <b>
                  checkbox
                </b>
                 - Enables custom style checkbox implementation
                 
              </li>
              <li>
                <b>
                  buttons
                </b>
                 - Predefined reusable button classes such as button--primary, button--disable etc
              </li>
              <li>
                <b>
                  confirmPopOverStyle
                </b>
                 - Specifies predefined common popover stylings
              </li>
            </ul>
            <h3>
              3. Style Css
            </h3>
            <ul>
              <li>
                <b>
                  index.css
                </b>
                 - Aggregator file of defined css files
              </li>
              <li>
                <b>
                  normaliza.css
                </b>
                 - Style file to reset default browser properties of HTML tags
              </li>
              <li>
                <b>
                  utilClasses
                </b>
                 - Style file contains common handy classes to put styling based on bootsrap class definitions
              </li>
              <li>
                <b>
                  fonts.css
                </b>
                imports font family required for project other than browser fonts
              </li>
            </ul>
          </div>
        </section>
        <section
          class="footer sc-bczRLJ eRSPWO"
        >
          <nav
            style="max-width: 1440px; margin: auto;"
          >
            <a
              class="footer-logo"
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
          </nav>
          <div
            class="disclaimer-text text-center"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
